lib/surface/formatter/phases/render.ex

```diff
@@ -113,8 +113,11 @@ defmodule Surface.Formatter.Phases.Render do
 
     expr =
       case expr do
-        [{:root, {:attribute_expr, expr, _meta}, __meta}] ->
-          Code.format_string!(expr)
+        [attr] ->
+          attr
+          |> render_attribute()
+          |> String.slice(1..-2)
+          |> String.trim()
```

Comment:
Excellent catch to use `render_attribute` here.

Can you walk me through the purpose of `String.slice(1..-2)` and `String.trim` here?

Reply:
I understand the `trim` will remove an initial `\n` and spaces. I think I agree with that for multi-line expressions. How does this not affect single-line expressions though?

Reply:
`render_attribute(...)` for a `:root` attribute returns the string wrapped in curlies, `String.slice(1..-2)` removes the surrounding curlies, and `String.trim()` removes any surrounding whitespace/newlines.

For a single line expression (i.e. `item <- @some_prop.items`), `render_attributes` returns:
`{item <- @some_prop.items}`

`String.slice(1..-2)` returns:
`item <- @some_prop.items`

and `String.trim()` returns the same thing unchanged as there's no surrounding whitespace:
`item <- @some_prop.items`

For the multi-line expression I used in the test, `render_attribute(...)` returns (whitespace as `.` and newlines as `\n`):
```
{\n
....item <- @some_prop.items,\n
....item.type == Some.Long.Complicated.Atom,\n
....value = item.some_item_property\n
..}
```

`String.slice(1..-2)` returns:
```
\n
....item <- @some_prop.items,\n
....item.type == Some.Long.Complicated.Atom,\n
....value = item.some_item_property\n

```

And finally `String.trim()` returns:
```
item <- @some_prop.items,\n
....item.type == Some.Long.Complicated.Atom,\n
....value = item.some_item_property
```

---

a better approach would probably be to extract out the bit that formats the code from `render_attribute` and have both functions call that instead of relying on the implementation detail that block expressions are currently treated as `:root` attributes; I mainly went this route because it was less modification and I'm not super familiar with this code base currently :-)

Reply:
Thanks for the explanation! I think this is perfect. It's pragmatic, and we can optimize later if we want.